aws-logs-subscriptionfilter/src/main/java/software/amazon/logs/subscriptionfilter/Translator.java

```diff
@@ -0,0 +1,121 @@
+import software.amazon.cloudformation.exceptions.*;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.HashSet;
+import java.util.HashMap;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class Translator {
+
+  public static BaseHandlerException translateException(final AwsServiceException e) {
+    if (e instanceof LimitExceededException) {
+      return new CfnServiceLimitExceededException(e);
+    }
+    if (e instanceof OperationAbortedException) {
+      return new CfnResourceConflictException(e);
+    }
+    if (e instanceof InvalidParameterException) {
+      return new CfnInvalidRequestException(e);
+    }
+    else if (e instanceof ResourceNotFoundException) {
+      return new CfnNotFoundException(e);
+    }
+    else if (e instanceof ServiceUnavailableException) {
+      return new CfnServiceInternalErrorException(e);
+    }
+    return new CfnGeneralServiceException(e);
+  }
+
+  static ResourceModel translateSubscriptionFilter
+          (final software.amazon.awssdk.services.cloudwatchlogs.model.SubscriptionFilter subscriptionFilter) {
+    return ResourceModel.builder()
+            .filterName(subscriptionFilter.filterName())
+            .destinationArn(subscriptionFilter.destinationArn())
+            .filterPattern(subscriptionFilter.filterPattern())
+            .logGroupName(subscriptionFilter.logGroupName())
+            .roleArn(subscriptionFilter.roleArn())
+            .build();
+  }
+
+  static software.amazon.awssdk.services.cloudwatchlogs.model.SubscriptionFilter translateToSDK
+          (final ResourceModel model) {
+    return software.amazon.awssdk.services.cloudwatchlogs.model.SubscriptionFilter.builder()
+            .filterName(model.getFilterName())
+            .destinationArn(model.getDestinationArn())
+            .filterPattern(model.getFilterPattern())
+            .logGroupName(model.getLogGroupName())
+            .roleArn(model.getRoleArn())
+            .build();
+  }
+
+  static PutSubscriptionFilterRequest translateToCreateRequest(final ResourceModel model) {
+    return PutSubscriptionFilterRequest.builder()
+            .filterName(model.getFilterName())
+            .destinationArn(model.getDestinationArn())
+            .filterPattern(model.getFilterPattern())
+            .logGroupName(model.getLogGroupName())
+            .roleArn(model.getRoleArn())
+            .build();
+  }
+
+  static DescribeSubscriptionFiltersRequest translateToReadRequest(final ResourceModel model) {
+    return DescribeSubscriptionFiltersRequest.builder()
+            .logGroupName(model.getLogGroupName())
+            .filterNamePrefix(model.getFilterName())
+            .build();
+  }
+
+  static ResourceModel translateFromReadResponse(final DescribeSubscriptionFiltersResponse awsResponse) {
+    return awsResponse.subscriptionFilters()
```

Comment:
Do we need match the filtername here? 
Since we are using filterNamePrefix while making read request, what happens if there are 2 filters with same prefix. 
e.g., `testFilter1` and `testFilter12`. My assumption is that, the describe will return both these filters.